From 94e93f5de1c62febdfd58078469b28c35eeaefc9 Mon Sep 17 00:00:00 2001
From: Alex Cruz <du.alexcruz@gmail.com>
Date: Mon, 26 Mar 2018 20:24:26 -0400
Subject: [PATCH 19/21] Expose and tint settings dashboard icon tint

Doing some research, I found that settings icons are tinted by using
colorControlNormal. While using our dark theme, we define colorControlNormal
and call on the system accent. Everything looks great using the dark theme.

The problem here was that the system was not defining it in the styles while
using the light(default) theme so the icon tint default to #737373 so when
you applied an accent using the light theme, those icons were not being changed.

This commit defines colorControlNormal for the Settings styles and also exposes
it so that we can change this with the accents, all while keeping the stock look
in place. Win win :-)

Change-Id: Ibfd150c31cc4404e95c5cab49e9ddf90124a9b14
---
 core/res/res/values/lineage_colors.xml        | 19 +++++++++++++++++++
 core/res/res/values/lineage_symbols.xml       |  5 ++++-
 .../res/res/values/themes_device_defaults.xml |  1 +
 3 files changed, 24 insertions(+), 1 deletion(-)
 create mode 100644 core/res/res/values/lineage_colors.xml

diff --git a/core/res/res/values/lineage_colors.xml b/core/res/res/values/lineage_colors.xml
new file mode 100644
index 00000000000..0b37094680e
--- /dev/null
+++ b/core/res/res/values/lineage_colors.xml
@@ -0,0 +1,19 @@
+<!--
+     Copyright (C) 2018 The LineageOS Project
+
+     Licensed under the Apache License, Version 2.0 (the "License");
+     you may not use this file except in compliance with the License.
+     You may obtain a copy of the License at
+
+          http://www.apache.org/licenses/LICENSE-2.0
+
+     Unless required by applicable law or agreed to in writing, software
+     distributed under the License is distributed on an "AS IS" BASIS,
+     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+     See the License for the specific language governing permissions and
+     limitations under the License.
+-->
+<resources>
+    <!-- Settings icon tint color -->
+    <color name="settings_icon_tint_color">#737373</color>
+</resources>
diff --git a/core/res/res/values/lineage_symbols.xml b/core/res/res/values/lineage_symbols.xml
index 0ce4b2d6f95..385f50953e6 100644
--- a/core/res/res/values/lineage_symbols.xml
+++ b/core/res/res/values/lineage_symbols.xml
@@ -1,7 +1,7 @@
 <!--
      Copyright (C) 2012-2015 The CyanogenMod Project
      Copyright (c) 2013, The Linux Foundation. All rights reserved.
-     Copyright (C) 2017 The LineageOS Project
+     Copyright (C) 2017-2018 The LineageOS Project
 
      Licensed under the Apache License, Version 2.0 (the "License");
      you may not use this file except in compliance with the License.
@@ -64,4 +64,7 @@
 
     <!-- Path to fast charging status file to detect whether an oem fast charger is active -->
     <java-symbol type="array" name="config_oemFastChargerStatusPaths" />
+
+    <!-- Settings icon tint color -->
+    <java-symbol type="color" name="settings_icon_tint_color" />
 </resources>
diff --git a/core/res/res/values/themes_device_defaults.xml b/core/res/res/values/themes_device_defaults.xml
index bf0c9066039..1571f9e0668 100644
--- a/core/res/res/values/themes_device_defaults.xml
+++ b/core/res/res/values/themes_device_defaults.xml
@@ -743,6 +743,7 @@ easier.
         <item name="colorPrimary">@color/primary_device_default_settings_light</item>
         <item name="colorPrimaryDark">@color/primary_dark_device_default_settings_light</item>
         <item name="colorSecondary">@color/secondary_device_default_settings_light</item>
+        <item name="colorControlNormal">@color/settings_icon_tint_color</item>
         <item name="colorAccent">@color/accent_device_default_light</item>
         <item name="colorEdgeEffect">@android:color/black</item>
 
-- 
2.17.1

